AEPMessaging/Sources/Messaging.swift

```diff
@@ -91,26 +98,33 @@ public class Messaging: NSObject, Extension {
         MessagingConstants.PersonalizationSchemas.RULESET_ITEM
     ]
 
+    private weak static var shared: Messaging?
+    static func getInstance() -> Messaging? {
+        shared
+    }
+
     // MARK: - Extension protocol methods
 
     public required init?(runtime: ExtensionRuntime) {
         self.runtime = runtime
         MessagingMigrator.migrate(cache: cache)
         rulesEngine = MessagingRulesEngine(name: MessagingConstants.RULES_ENGINE_NAME, extensionRuntime: runtime, cache: cache)
-        feedRulesEngine = FeedRulesEngine(name: MessagingConstants.FEED_RULES_ENGINE_NAME, extensionRuntime: runtime)
+        contentCardRulesEngine = ContentCardRulesEngine(name: MessagingConstants.CONTENT_CARD_RULES_ENGINE_NAME, extensionRuntime: runtime)
         super.init()
         loadCachedPropositions()
+        Messaging.shared = self
     }
 
     /// INTERNAL ONLY
     /// used for testing
-    init(runtime: ExtensionRuntime, rulesEngine: MessagingRulesEngine, feedRulesEngine: FeedRulesEngine, expectedSurfaceUri _: String, cache: Cache) {
+    init(runtime: ExtensionRuntime, rulesEngine: MessagingRulesEngine, contentCardRulesEngine: ContentCardRulesEngine, expectedSurfaceUri _: String, cache: Cache) {
         self.runtime = runtime
         self.rulesEngine = rulesEngine
-        self.feedRulesEngine = feedRulesEngine
+        self.contentCardRulesEngine = contentCardRulesEngine
         self.cache = cache
         super.init()
         loadCachedPropositions()
+        Messaging.shared = self
```

Comment:
Let's create a list of all usages of the extension runtime to understand if we can decouple the extension instance of the various levels of functionality required in the other classes, e.g. is it just to dispatch events, is it for shared state access, anything else?